test/Release/MysqlReleaseToggleCollectionGatewayTest.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: Volodymyr
+ * Date: 12/08/2015
+ * Time: 13:13
+ */
+
+namespace Clearbooks\LabsMysql\Release;
+
+
+use Doctrine\DBAL\Configuration;
+use Doctrine\DBAL\DriverManager;
+use Clearbooks\LabsMysql\Toggle\Entity\Toggle;
+
+class MysqlReleaseToggleCollectionGatewayTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var MysqlReleaseToggleCollectionGateway
+     */
+    private $gateway;
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $connectionParams = array(
+            'dbname' => 'labs',
+            'user' => 'root',
+            'password' => '',
+            'host' => 'localhost',
+            'driver' => 'pdo_mysql',
+        );
+
+        $this->connection = DriverManager::getConnection( $connectionParams, new Configuration() );
+        $this->gateway = new MysqlReleaseToggleCollectionGateway( $this->connection );
+    }
+
+    /**
+     * @test
+     */
+    public function givenNoExistentRelease_ReleaseToggleColection_ReturnsEmptyArray()
+    {
+        $returnedToggles = $this->gateway->getTogglesForRelease( 'bloop' );
+        $this->assertEquals( [ ], $returnedToggles );
+    }
+
+    /**
+     * @test
+     */
+    public function givenNoExistentTogglesInTheExistentRelase_ReleaseToggleColection_ReturnsEmptyArray()
+    {
+        $releaseName = 'Test release for toggle 1';
+        $url = 'a helpful url';
+        $id = $this->addRelease( $releaseName, $url );
+
+        $returnedToggles = $this->gateway->getTogglesForRelease( $id );
+
+        // Teardown
+        $this->deleteAddedRelease( $id );
+
+        $this->assertEquals( [ ], $returnedToggles );
+    }
+
+    /**
+     * @test
+     */
+    public function givenExistentTogglesInTheExistentRelease_ReleaseToggleColoction_ReturnsArrayOfExistentToggles()
+    {
+        $releaseName = 'Test release for toggle 2';
+        $url = 'a helpful url';
+        $id = $this->addRelease( $releaseName, $url );
+
+        $toggleId1 = $this->addToggle( "test1", $id );
+        $toggleId2 = $this->addToggle( "test2", $id );
+
+        $expectedToggle = new Toggle( "test1", $id );
+        $expectedToggle2 = new Toggle( "test2", $id );
+
+        $expectedToggles = [ $expectedToggle, $expectedToggle2 ];
+        $returnedToggles = $this->gateway->getTogglesForRelease( $id );
+
+        // Teardown
+        $this->deleteAddedToggle( $toggleId1 );
+        $this->deleteAddedToggle( $toggleId2 );
+        $this->deleteAddedRelease( $id );
+
+        $this->assertEquals( $expectedToggles, $returnedToggles );
+
+        foreach ( $expectedToggles as $key => $value ) {
```

Comment:
Teardown is placed above the assert in this instance to avoid unreasonable mess in the database if test fails.
+ this way you can keep testing.
- if not you will have to manually re configure and clean database every time you fail the test.. which is often in TDD!